src/main/java/edu/harvard/iq/dataverse/search/SearchIncludeFragment.java

```diff
@@ -1495,7 +1495,15 @@ private DataverseRequest getDataverseRequest() {
     }
 
     public boolean isValid(SolrSearchResult result) {
+        Long id = result.getEntityId();
+        
+        DvObject obj = dvObjectService.findDvObject(id);
+        if(obj != null && obj instanceof Dataset) {
+                
         return result.isValid(x -> permissionsWrapper.canUpdateDataset(getDataverseRequest(), datasetService.find(x.getEntityId())));
```

Comment:
Passing the function instead of boolean indicating if the user has an update permission is for optimization reasons, where the cheap checks are done first (e.g., the dataset is valid and the boolean is already in the search result, which should cover most common cases), before a query is executed to find the DV object. The query and the permission check happen only when the search object is a dataset with incomplete metadata, is not draft and the boolean for displaying incomplete labels for published datasets is set to true, which should be an uncommon situation. I am not sure where the nullpointer comes from, but putting the null check inside the lambda should be way more efficient, something like this:

```java
    public boolean isValid(SolrSearchResult result) {
        return result.isValid(x -> {
            Long id = x.getEntityId();
            DvObject obj = dvObjectService.findDvObject(id);
            if(obj != null && obj instanceof Dataset) {
                return permissionsWrapper.canUpdateDataset(getDataverseRequest(), (Dataset) obj);
            }
            logger.fine("isValid called for dvObject that is null (or not a dataset), id: " + id + "This can occur if a dataset is deleted while a search is in progress");
            return true;
        });
    }
```

Reply:
change made - thanks!